### PR TITLE
[482] integrating omniauth and dfe sign in

### PR DESCRIPTION
**Pull request by defong**
### Context
integrating omniauth and dfe sign in

### Changes proposed in this pull request
integrating omniauth and dfe sign in

### Guidance to review
Best to run the code, configure the required settings for DfE Sign-in and login

`use_dfe_sign_in` is default to true as the `persona` is in an incompatibility state, a follow up PR from @JonF1 is in flight

To easy resistance, a feature flag `allow_user_creation` has been added, its function is save the `non existence user` to the database on the provider `DfE`, the provider `DfE` will also be created if it is `non existence`

### CODE REVIEW

**Comment by defong** [comment]
[[L] Integrating OmniAuth and DfE Sign In](https://trello.com/c/twqbASbr/482-l-integrating-omniauth-and-dfe-sign-in)

**Review by aliuk2012 — APPROVED** [review]
### Tested this locally and it WORKED 😄 👍

File: config/settings.yml [hunk 1]
```diff
@@ -12,3 +13,11 @@ dttp:
 features:
   home_text: false
   use_ssl: true
+  use_dfe_sign_in: true
+  allow_user_creation: false
+
+dfe_sign_in:
+  identifier: identifier required value
```
Comment (dankmitchell) [inline config/settings.yml:20]:
```suggestion
  identifier: rtt
```

File: config/settings.yml [hunk 2]
```diff
@@ -12,3 +13,11 @@ dttp:
...
+  issuer: issuer required value
```
Comment (dankmitchell) [inline config/settings.yml:21]:
```suggestion
  issuer: https://test-oidc.signin.education.gov.uk/
```

File: config/settings.yml [hunk 3]
```diff
@@ -12,3 +13,11 @@ dttp:
...
+  profile: profile required value
```
Comment (dankmitchell) [inline config/settings.yml:22]:
```suggestion
  profile: https://test-profile.signin.education.gov.uk
```

File: config/settings.yml [hunk 4]
```diff
@@ -1,4 +1,5 @@
 port: 5000
+base_url: base_url required value
```
Comment (dankmitchell) [inline config/settings.yml:2]:
```suggestion
base_url: https://localhost:5000
```

File: config/settings/development.yml [hunk 5]
```diff
@@ -1,3 +1,9 @@
 basic_auth_required: false
 dttp:
   api_base_url: "https://dttp-dev.api.crm4.dynamics.com"
+
+base_url: https://localhost:5000
```
Comment (dankmitchell) [inline config/settings/development.yml:5]: This can be removed

**Pull request merged**